pkg/controller/nstemplateset/nstemplateset_controller.go

```diff
@@ -68,48 +81,70 @@ func (r *ReconcileNSTemplateSet) Reconcile(request reconcile.Request) (reconcile
 		}
 		return reconcile.Result{}, err
 	}
-	err = r.ensureNamespaces(reqLogger, nsTmplSet)
-	return reconcile.Result{}, err
+	return r.ensureNamespaces(reqLogger, nsTmplSet)
 }
 
-func (r *ReconcileNSTemplateSet) ensureNamespaces(logger logr.Logger, nsTmplSet *toolchainv1alpha1.NSTemplateSet) error {
-	name := nsTmplSet.GetName()
+func (r *ReconcileNSTemplateSet) ensureNamespaces(logger logr.Logger, nsTmplSet *toolchainv1alpha1.NSTemplateSet) (reconcile.Result, error) {
+	userName := nsTmplSet.GetName()
 	labels := map[string]string{
-		"owner": name,
+		"owner": userName,
 	}
 	opts := client.MatchingLabels(labels)
-	projects := &projectv1.ProjectList{}
-	if err := r.client.List(context.TODO(), opts, projects); err != nil {
+	namespaces := &corev1.NamespaceList{}
+	if err := r.client.List(context.TODO(), opts, namespaces); err != nil {
 		// TODO status handling
-		return err
+		return reconcile.Result{}, err
 	}
 
-	nsForProvision := findNsForProvision(projects.Items, nsTmplSet.Spec.Namespaces, name)
-	for _, ns := range nsForProvision {
+	missingNs := findNsForProvision(namespaces.Items, nsTmplSet.Spec.Namespaces, userName)
+	if missingNs != (toolchainv1alpha1.Namespace{}) {
 		// TODO call template processing for ns
-		log.Info("provisionng namespace", "namespace", ns)
+		log.Info("provisioning namespace", "namespace", missingNs)
+
+		// set labels
+		nsName := toNamespaceName(userName, missingNs.Type)
+		namespace := &corev1.Namespace{}
+		if err := r.client.Get(context.TODO(), types.NamespacedName{Name: nsName}, namespace); err != nil {
+			return reconcile.Result{}, err
+		}
```

Comment:
@dipak-pawar as discussed, I assumed that when a call returned from `Template.ProcessAndApply()` the `Project.Status.Phase` will be `Active` (in positive scenario). In turns, I can also assume that to check `Project` status, you would have make `client.Get(project)` call successful.  After `Template.ProcessAndApply()`, I can safely call `r.client.Get(namespace)` as above to set labels.

Reply:
@dipak-pawar @xcoulon I see this is the case in `Template.ProcessAndApply` https://github.com/codeready-toolchain/member-operator/pull/49/files?file-filters%5B%5D=.go#diff-ae1e62ddb6744c2036a1b470ddc6a527R88
Thanks :-)